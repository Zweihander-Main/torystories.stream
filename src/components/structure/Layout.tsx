import React from 'react';
import Footer from 'components/structure/Footer';
import { RiHome2Line } from 'react-icons/ri';
import { Link } from 'gatsby';

type LayoutProps = {
	showHomeButton?: boolean;
	hideFooter?: boolean;
};

const Layout: React.FC<
	LayoutProps & React.PropsWithChildren<Record<string, unknown>>
> = ({ children, showHomeButton = false, hideFooter = false }) => {
	return (
		<>
			{!showHomeButton && (
				<noscript className="static top-0 w-full text-center text-white bg-deepPurple">
					This site will work without JavaScript but interactive
					elements like the podcast player will not.{' '}
					<a href="https://github.com/Zweihander-Main/torystories.stream">
						Source code available.
					</a>
				</noscript>
			)}
			<div className="flex-col items-center mb-20 text-white bg-black rflex layout font-body">
				{showHomeButton && (
					<nav
						role="navigation"
						aria-label="Home"
						className={'absolute left-0 top-2 xl:top-4 z-50'}
					>
						<Link
							to={'/'}
							className={
								'cursor-pointer relative px-2 pt-3 pb-2 lg:-top-1 xl:p-4 no-underline lg:shadow-navButton bg-black bg-opacity-40 rounded-1xl'
							}
						>
							<RiHome2Line className="inline text-xl lg:text-2xl xl:text-3xl " />
							<span
								className={
									'hidden lg:inline font-display tracking-display pl-2 relative top-1 text-xl xl:text-2xl text-shadow-md'
								}
							>
								Home
							</span>
						</Link>
					</nav>
				)}
				<main>{children}</main>
				{!hideFooter && <Footer />}
			</div>
		</>
	);
};

export default Layout;
