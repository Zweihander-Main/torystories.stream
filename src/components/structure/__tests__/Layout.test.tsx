import * as React from 'react';
import { render } from '@testing-library/react';
import Layout from '../Layout';

describe('Layout', () => {
	it('preserves its structure', () => {
		expect(render(<Layout />).container).toMatchSnapshot();
	});
});
