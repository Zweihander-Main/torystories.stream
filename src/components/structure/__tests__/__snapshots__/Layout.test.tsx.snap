// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
<div>
  <div
    class="layout flex flex-col items-center bg-black text-white font-body mb-20"
  >
    <main />
    <footer
      class="m-20 text-center"
    >
      <p>
        Check out:
      </p>
      <p
        class="mb-4"
      >
        <a
          class="font-bold"
          href="https://www.tbwns.com"
        >
          Martin's blog
        </a>
         | 
        <a
          class="font-bold"
          href="https://www.lordliverpool.com"
        >
          His biography of Lord Liverpool
        </a>
      </p>
      <p>
        Â© 
        2021
         Martin Hutchinson
      </p>
    </footer>
  </div>
</div>
`;
